FESS-GUI/flywheeloperation.cpp

```diff
@@ -1,58 +1,258 @@
+// Custom Libraries
 #include "flywheeloperation.h"
-#include <cmath>
-#include <QTime>
-#include <QtMath>
 
-FlywheelOperation::FlywheelOperation()
+FlywheelOperation::FlywheelOperation(Interface *inter)
 {
     upperDisplacement = new QPointF();
     lowerDisplacement = new QPointF();
     rotationalPosition = new QPointF();
+
+    interface = inter;
+    setDefaults();
 }
 
-void FlywheelOperation::setMotion(float velocity = NAN, float acceleration = NAN, float jerk = NAN)
+void FlywheelOperation::setMotion(float velocity, float acceleration, float jerk)
 {
+    interface->pushCommand(COMMAND_SET_ALLD_FLOA);
+    interface->pushData(velocity);
+    interface->pushData(acceleration);
+    interface->pushData(jerk);
+}
 
+void FlywheelOperation::setVelocity(float velocity)
+{
+    interface->pushCommand(COMMAND_SET_VELO_FLOA);
+    interface->pushData(velocity);
+}
+
+void FlywheelOperation::setAcceleration(float acceleration)
+{
+    interface->pushCommand(COMMAND_SET_ACCE_FLOA);
+    interface->pushData(acceleration);
+}
+
+void FlywheelOperation::setJerk(float jerk)
+{
+    interface->pushCommand(COMMAND_SET_JERK_FLOA);
+    interface->pushData(jerk);
 }
 
 float FlywheelOperation::getVelocity()
 {
-    double key = QDateTime::currentDateTime().toMSecsSinceEpoch()/1000.0;
-    return qSin(key*1.6+qCos(key*1.7)*2)*10 + qSin(key*1.2+0.56)*20 + 26;
+    float val = vel.front();
```

Comment:
I don't feel safe assuming the GUI sample rate will exceed the fastest sampling rate in the controller. When I was running the GUI on my laptop just now, it was running at 24 FPS. If the GUI ran at less than 20 FPS, that would be sampling less than every 50 ms, right? Wouldn't that result in values getting stacked up in the queue, and the graphs not showing the most recent values?